staging/src/k8s.io/component-base/cli/run.go

```diff
@@ -42,8 +59,8 @@ import (
 //
 // Commands like kubectl where logging is not normally part of the runtime output
 // should use RunNoErrOutput instead and deal with the returned error themselves.
-func Run(cmd *cobra.Command) int {
-	if logsInitialized, err := run(cmd); err != nil {
+func Run(cmd *cobra.Command, opts ...RunOption) int {
```

Comment:
Run gets called in too many places to add a mandatory parameter. Functional options are added because then code which just uses `cli,Run(cmd)` continues to work unchanged.

Reply:
Another alternative: instead of functional options, cli and logs could have their own, private package-level `FeatureGate` instance, initially nil. The API for setting that would be something like `cli.UseFeatureGate` which remembers the passed in `FeatureGate` instance and adds the package's feature gates to it. It needs to call the corresponding function of other packages that it uses, in this case `logs.UseFeatureGate`, because the user of `cli` might not know about those indirect dependencies and/or those might change.

`cli.UseFeatureGate` would have to be called during sequential parts of the code because it registers feature gates in `FeatureGate` which (in the current implementation) doesn't protect itself against concurrent access, so adding a mutex in `cli.UseFeatureGate` alone wouldn't be insufficient.

I'm undecided whether this is better. Thoughts?